### PR TITLE
Enable dep-approvers to take care of go.mod/sum for k8s.io/api

### PR DESCRIPTION
**Pull request by dims**
A bunch of golang changes and dependency updates end up touching go.mod/sum under k8s.io/api. These can be easily taken care of by dependency approvers and does not need api approval necessarily.

the rest of the staging repos have `no_parent_owners: true` deeper in the hierarchy, so we just need to fix this instance.

/assign @thockin 

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: staging/src/k8s.io/api/OWNERS [hunk 1]
```diff
@@ -13,3 +13,7 @@ filters:
   "\\.go$":
     labels:
       - kind/api-change
+  # allow dependency approver(s) for just go.mod/go.sum files.
```
Comment (BenTheElder) [inline staging/src/k8s.io/api/OWNERS:16]: note that prow doesn't actually have per-file approvers implemented AFAIK

Comment (thockin) [reply staging/src/k8s.io/api/OWNERS:16]: So is this all moot?

Comment (dims) [reply staging/src/k8s.io/api/OWNERS:16]: Damn, i keep forgetting @BenTheElder!

**Pull request closed without merging**